yellowbrick/features/base.py

```diff
@@ -309,3 +312,130 @@ def _determine_target_color_type(self, y):
                 "could not determine target color type "
                 "from target='{}' to '{}'"
             ).format(self.target_type, self._target_color_type))
+
+
+##########################################################################
+## Projection Visualizers
+##########################################################################
+
+class ProjectionVisualizer(DataVisualizer):
+    
+    def __init__(self, transformer, ax=None, features=None, classes=None, color=None,
+             colormap=None, target_type="auto", proj_dim=2, alpha=0.75, **kwargs):
+        super(ProjectionVisualizer, self).__init__(ax=ax, features=features, 
+                                                     classes=classes, color=color,
+                                                     colormap=colormap, 
+                                                     target_type=target_type, **kwargs)
+        self.transformer=transformer
+        if proj_dim not in [2, 3]:
+            raise YellowbrickValueError("Projection dimensions must be either 2 or 3")
+        self.proj_dim = proj_dim
+        self.alpha = alpha
+        self._cax = None
+
+    @property
+    def cax(self):
+        """
+        The axes of the colorbar, right of the scatterplot.
+        """
+        if self._cax is None:
+            raise AttributeError(
+                "this visualizer does not have a histogram for the X axis"
+            )
+        return self._cax       
+        
+    def _layout_colorbar(self):
+        """
+        Creates the layout for colorbar. The colorbar is added to the right of the 
+        scatterplot 
+        """
+        # Ensure matplotlib version compatibility
+        if make_axes_locatable is None:
+            raise YellowbrickValueError((
+                "heatmap requires matplotlib 2.0.2 or greater "
+                "please upgrade matplotlib or set heatmap=False on the visualizer"
+            ))
```

Comment:
Please update the text of this exception.

Reply:
Although, see https://github.com/DistrictDataLabs/yellowbrick/issues/826#issuecomment-509249546 -- we may just want to make matplotlib v2.0.2 the minimum version and we can take out these warnings.

Reply:
I will remove this!

Reply:
Before you reset the minimum matplotlib version, let's get @rebeccabilbro and @DistrictDataLabs/team-oz-maintainers to chime in on the major version deprecation. Thoughts, guys?

Reply:
I'm at peace with this; it'll make #826 a heck of a lot easier to move forward with.